src/Talis/Persona/Client/Base.php

```diff
@@ -34,22 +39,63 @@
     private $httpClient;
 
     /**
+     * @var \Doctrine\Common\Cache\CacheProvider
+     */
+    private $cacheBackend;
+
+    /**
+     * @var string
+     */
+    private $keyPrefix;
+
+    /**
+     * @var int
+     */
+    private $defaultTtl;
+
+    /**
      * Constructor
      *
      * @param array $config An array of options with the following keys: <pre>
      *      persona_host: (string) the persona host you'll be making requests to (e.g. 'http://localhost')
      *      persona_oauth_route: (string) the token api route to query ( e.g: '/oauth/tokens')
-     *      tokencache_redis_host: (string) the host address of redis token cache
-     *      tokencache_redis_port: (integer) the port number the redis host ist listening
-     *      tokencache_redis_db: (integer) the database to connnect to</pre>
-     * @param logger the logger to use, otherwise a default will be assigned and used
+     *      cacheBackend: (Doctrine\Common\Cache\CacheProvider) optional cache storage (defaults to Filesystem)
+     *      cacheKeyPrefix: (string) optional prefix to append to the cache keys
+     *      cacheDefaultTTL: (integer) optional cache TTL value
+     * @param @deprecated $logger optional logger instance
+     * @param @deprecated $cacheBackend optional cache backend see
+     *  https://github.com/doctrine/cache/tree/master/lib/Doctrine/Common/Cache
+     *  https://doctrine-orm.readthedocs.org/projects/doctrine-orm/en/latest/reference/caching.html
      * @throws \InvalidArgumentException if any of the required config parameters are missing
+     * 
      */
-    public function __construct($config,\Psr\Log\LoggerInterface $logger=null) {
-        if($this->checkConfig($config)){
-            $this->config = $config;
-        };
-        $this->logger = $logger;
+    public function __construct(
+        array $config,
+        \Psr\Log\LoggerInterface $logger = null,
+        CacheProvider $cacheBackend = null
+    ) {
+        $this->checkConfig($config);
+        $this->config = $config;
+
+        $this->logger = isset($config['logger'])
+            ? $config['logger']
+            : $logger;
```

Comment:
Also, and this is just a personal preference, if you're going to write a shorthand if statement across multiple lines, it might as well be a full if/else statement.